cpp/include/cudf/datetime.hpp

```diff
@@ -17,117 +17,165 @@
 #ifndef DATETIME_HPP
 #define DATETIME_HPP
 
-#include "cudf.h"
-#include <functional>
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_view.hpp>
+#include <functional>
+#include "cudf.h"
 
-#include <rmm/rmm.hpp>
+#include <rmm/mr/default_memory_resource.hpp>
 
 namespace cudf {
 namespace datetime {
 namespace detail {
 
-  /**
-   * @brief Compare the input time units and returns whichever is more granular,
-   * i.e. the gdf_time_unit that can represent the highest precision timestamps.
-   * 
-   * @returns gdf_time_unit the more granular of the two input gdf_time_units.
-   */
-  gdf_time_unit common_resolution(gdf_time_unit lhs_unit, gdf_time_unit rhs_unit);
+enum class datetime_component {
+  invalid = 0,
```

Comment:
These should probably be UPPERCASE for consistency with the rest of libcudf.

Reply:
These are still lowercase...

Reply:
@harrism possible bug in the github UI? Latest version in my branch shows uppercase: https://github.com/trxcllnt/cudf/blob/libcudf%2B%2B/datetime_ops/cpp/include/cudf/datetime.hpp#L27

Reply:
Now it's showing the correct file for me. Maybe we had to wait for a github HTTP cache to expire.

Reply:
It still shows this code for me:

```
enum class DATETIME_COMPONENT {
   invalid = 0,
   year, 
   month,
   ...
```

What I expect is this:

```
enum class datetime_component {
   INVALID = 0,
   YEAR,
   MONTH,
   DAY
   ...
```

Is that what you see?

Reply:
Oh I misunderstood the meaning of your original comment :sweat_smile:. Should be fixed now.